From: Andrea Arcangeli <aarcange@redhat.com>
Date: Mon, 1 Feb 2010 15:17:22 -0500
Subject: [mm] backport page_referenced microoptimization
Message-id: <20100201152039.859954967@redhat.com>
Patchwork-id: 23032
O-Subject: [RHEL6 25/37] backport page_referenced microoptimization
Bugzilla: 556572
RH-Acked-by: Larry Woodman <lwoodman@redhat.com>

From: Andrea Arcangeli <aarcange@redhat.com>

Don't modify vm_flags if the vma didn't map the page.

Signed-off-by: Andrea Arcangeli <aarcange@redhat.com>
Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/mm/rmap.c b/mm/rmap.c
index 5aaf5d0..1bece3d 100644
--- a/mm/rmap.c
+++ b/mm/rmap.c
@@ -384,9 +384,10 @@ int page_referenced_one(struct page *page, struct vm_area_struct *vma,
 out_unmap:
 	(*mapcount)--;
 	pte_unmap_unlock(pte, ptl);
-out:
+
 	if (referenced)
 		*vm_flags |= vma->vm_flags;
+out:
 	return referenced;
 }
 
